perfkitbenchmarker/benchmark_spec.py

```diff
@@ -153,90 +163,109 @@ def UnPickleLock(locked, *args):
-      self.image = []
-      self.machine_type = []
-      for node in self.config.node_sections:
-        self.vm_dict[node.split(':')[1]] = []
-      args = [((config_dict[node],
-                node.split(':')[1]), {}) for node in self.config.node_sections]
-      vm_util.RunThreaded(
-          self.CreateVirtualMachineFromNodeSection, args)
-      self.num_vms = len(self.vms)
-      self.image = ','.join(self.image)
-      self.zones = ','.join(self.zones)
-      self.machine_type = ','.join(self.machine_type)
-    else:
-      self.cloud = FLAGS.cloud
-      self.project = FLAGS.project
-      self.zones = FLAGS.zones
-      self.image = FLAGS.image
-      self.machine_type = FLAGS.machine_type
-      if benchmark_info['num_machines'] is None:
-        self.num_vms = FLAGS.num_vms
-      else:
-        self.num_vms = benchmark_info['num_machines']
-      self.scratch_disk = benchmark_info['scratch_disk']
-      self.scratch_disk_size = FLAGS.scratch_disk_size
-      self.scratch_disk_type = FLAGS.scratch_disk_type
-      self.scratch_disk_iops = FLAGS.scratch_disk_iops
-
-      self.vms = [
-          self.CreateVirtualMachine(
-              self.zones[min(index, len(self.zones) - 1)])
-          for index in range(self.num_vms)]
-      self.vm_dict['default'] = self.vms
-      for vm in self.vms:
-        # If we are using local disks and num_striped_disks has not been
-        # set, then we want to set it to stripe all local disks together.
-        if (FLAGS.scratch_disk_type == disk.LOCAL and
-            benchmark_info['scratch_disk'] and
-            not FLAGS['num_striped_disks'].present):
-          num_striped_disks = (vm.max_local_disks //
-                               benchmark_info['scratch_disk'])
-          if num_striped_disks == 0:
-            raise errors.Error(
-                'Not enough local disks to run benchmark "%s". It requires at '
-                'least %d local disk(s). The specified machine type has %d '
-                'local disk(s).' % (benchmark_info['name'],
-                                    int(benchmark_info['scratch_disk']),
-                                    vm.max_local_disks))
-        else:
-          num_striped_disks = FLAGS.num_striped_disks
-        for i in range(benchmark_info['scratch_disk']):
-          mount_point = '%s%d' % (FLAGS.scratch_dir, i)
-          disk_spec = disk.BaseDiskSpec(
-              self.scratch_disk_size, self.scratch_disk_type,
-              mount_point, self.scratch_disk_iops,
-              num_striped_disks)
-          vm.disk_specs.append(disk_spec)
-
-    self.file_name = '%s/%s' % (vm_util.GetTempDir(), benchmark_info['name'])
+    self.vm_groups = {}
     self.deleted = False
+    self.file_name = os.path.join(vm_util.GetTempDir(), self.name)
     self.always_call_cleanup = False
 
+  def _GetCloudForGroup(self, group_name):
+    """Gets the cloud for a VM group by looking at flags and the config."""
```

Comment:
Is precedence specified somewhere? flag > config override (?) > user config (?) > config